packages/expo-constants/scripts/source-login-scripts.sh

```diff
@@ -9,7 +9,12 @@
 
 current_shell=$(ps -cp "$$" -o comm='' | sed 's/^-//')
 
-if [[ "$current_shell" == zsh ]]; then
+export NVM_DIR=$HOME/.nvm;
+NVM_SH_FILE=$NVM_DIR/nvm.sh
+# first attempt to source via nvm and fallback to sourcing via zsh or bash
+if test -f "$NVM_SH_FILE"; then
+  source $NVM_SH_FILE
+elif [[ "$current_shell" == zsh ]]; then
```

Comment:
```suggestion
# attempt to source via nvm
if test -f "$HOME/.nvm/nvm.sh"; then
  export NVM_DIR=$HOME/.nvm
  source $NVM_DIR/nvm.sh

if [[ "$current_shell" == zsh ]]; then
```

two thought for this suggestion:
  1. if `$HOME/.nvm/nvm.sh` doesn't exist, we should not export NVM_DIR.
  2. sourcing zsh or bash doesn't necessary to be fallback. we can source .env, source nvm.sh, and source shell rc separately.

Reply:
Thanks for the feedback! I updated - could you take a look @Kudo?

> if $HOME/.nvm/nvm.sh doesn't exist, we should not export NVM_DIR.
I went ahead and removed the `NVM_DIR` export altogether. I don't think it's needed since we are just referencing it in this script and not elsewhere. Lmk if you think otherwise.

> sourcing zsh or bash doesn't necessary to be fallback. we can source .env, source nvm.sh, and source shell rc separately.
Ah yes that makes sense


> i think we can do the nvm support first and i'll try to follow up with the .env later.

that sounds good to me